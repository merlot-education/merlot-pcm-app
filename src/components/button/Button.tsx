import React from 'react';
import { ViewStyle } from 'react-native';

import AntButton from '@ant-design/react-native/lib/button';

export enum ButtonType {
  Primary = 'primary',
  Warning = 'warning',
  Ghost = 'ghost',
}

interface ButtonProps {
  title: string;
  buttonType: ButtonType;
  onPress?: () => void;
  disabled?: boolean;
  buttonStyle?: ViewStyle;
}

const Button: React.FC<ButtonProps> = ({
  title,
  buttonType = ButtonType.Primary,
  buttonStyle,
  onPress,
  disabled = false,
}) => {
  return (
    <AntButton
      testID="button"
      accessibilityLabel={title}
      type={buttonType}
      style={buttonStyle}
      disabled={disabled}
      onPress={onPress}>
      {title}
    </AntButton>
  );
};

export default Button;
