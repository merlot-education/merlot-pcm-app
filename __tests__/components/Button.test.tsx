import React from 'react'
import { create } from 'react-test-renderer'

import Button, { ButtonType } from '../../src/components/button/Button'

describe('Button Component', () => {
  it('Primary renders correctly', () => {
    const tree = create(
      <Button
        title="Hello Primary"
        onPress={() => {}}
        buttonType={ButtonType.Primary}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Secondary renders correctly', () => {
    const tree = create(
      <Button
        title="Hello Secondary"
        onPress={() => {}}
        buttonType={ButtonType.Secondary}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
