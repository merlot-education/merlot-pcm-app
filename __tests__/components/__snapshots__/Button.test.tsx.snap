// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Primary renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "borderWidth": 1,
        "justifyContent": "center",
      },
      Object {
        "height": 47,
        "paddingLeft": 15,
        "paddingRight": 15,
      },
      Object {
        "backgroundColor": "#108ee9",
        "borderColor": "#108ee9",
      },
      false,
      false,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
          },
          Object {
            "color": "#ffffff",
          },
          false,
          false,
        ]
      }
    >
      Hello Primary
    </Text>
  </View>
</View>
`;
